app/controllers/mixins/actions/physical_server_actions/misc.rb

```diff
@@ -0,0 +1,66 @@
+module Mixins
+  module Actions
+    module PhysicalServerActions
+      module Misc
+        def process_physical_servers_refresh(physical_servers, task, display_name)
+          PhysicalServer.refresh_ems(physical_servers)
+          add_flash(n_("%{task} initiated for %{count} Physical Server from the %{product} Database",
+                       "%{task} initiated for %{count} Physical Servers from the %{product} Database", physical_servers.length) % \
+            {:task    => display_name,
+             :product => I18n.t('product.name'),
+             :count   => physical_servers.length})
+          AuditEvent.success(:userid => session[:userid], :event => "physical_server_#{task}",
+              :message => "'#{task_name(task)}' successfully initiated for #{pluralize(physical_servers.length, "PhysicalServer")}",
+              :target_class => "PhysicalServer")
+        end
+
+        def process_physical_servers(physical_servers, task, display_name = nil)
+          physical_servers, _physical_servers_out_region = filter_ids_in_region(physical_servers, _("PhysicalServer"))
+          return if physical_servers.empty?
+
+          display_name ||= task
```

Comment:
why this line?

jsut make `display_name` obligatory 

if you don't and the task ever get's into display_name then in `process_physical_servers_refresh` the flash message will not be properly internationalized

you probably followed some bad pattern again, but seems you don't need this here.